site/src/pages/release-notes/1.3.0.mdx

```diff
@@ -0,0 +1,231 @@
+  annotated service. #3088 #3124, #3192
+  - Use `armeria-protobuf`, `armeria-scalapb_2.12` or `armeria-scalapb_2.13` dependencies.
+  - See [Supporting ScalaPB in annotated services](/docs/advanced-scalapb#supporting-scalapb-in-annotated-services)
+    for more information.
+- You can now use Scala `Future` in annotated service. #3189
+  ```scala
+  @Get("/items/{id}")
+  def items(@Param id: Int)(implicit ec: ExecutionContext): Future[String] = {
+    Future {
+      // Perform asynchronous task using Armeria's event loop.
+      ...
+    }
+  }
+  ```
+- You can now use <type://HttpDeframer> to conveniently decode a stream of <typeplural://HttpObject>
+  to N objects. #2981
+  ```java
+  HttpDeframerHandler<String> decoder = ...
+  HttpDeframer<String> deframer = new HttpDeframer<>(decoder, ByteBufAllocator.DEFAULT);
+  HttpRequest request = ...;
+  request.subscribe(deframer);
+  ```
+  See <type://HttpDeframer> for more information.
+- You can now apply <type://CircuitBreaker> per request path. #3134, #3135
+  ```java
+  CircuitBreakerFactory factory = ...
+  CircuitBreakerMapping mapping = CircuitBreakerMapping.builder()
+                                                       .perPath() // per the combination of host and path
+                                                       .perHost()
+                                                       .build(factory)
+  CircuitBreakerRule rule = ...
+  CircuitBreakerClient.newDecorator(mapping, rule);
+  ```
+- You can now apply the different `maxTotalAttempts` and `responseTimeout` for <type://RetryingClient>
+  using <type://RetryConfig>. #3145
+  ```java
+  BiFunction<ClientRequestContext, Request, String> keyFactory =
+      (ctx, req) -> ctx.endpoint().host();
+  BiFunction<ClientRequestContext, Request, RetryConfig<HttpResponse>> configFactory = (ctx, req) -> {
+      String host = ctx.endpoint().host();
+      RetryConfigBuilder builder = RetryConfig.<HttpResponse>builder(RetryRule.onException());
+      if (host.equals("host1")) {
+          builder.maxTotalAttempts(2);
+      } else if (host.equals("host2")) {
+          builder.maxTotalAttempts(4);
+      } else {
+          builder.maxTotalAttempts(1);
+      }
+      return builder.build();
+  };
+  RetryConfigMapping mapping = RetryConfigMapping.of(keyFactory, configFactory);
+  RetryingClient.newDecoratorWithMapping(mapping);
+  ```
+- You can now split the <type://ResponseHeaders> and bodies using <type://HttpResponse#split()>. #3038
+- You can now specify the Caffeine spec for the DNS resolver. #2970, #3007
+- You can now specify a prefix fo MDC keys using the `<prefix>` element. #3086, #3112
+  ```xml
+  <prefix>...</prefix> <!-- The default prefix -->
+  <export>...</export>
+  <exports>...</exports>
+
+  <exportGroup>
+    <prefix>...</prefix> <!-- The prefix for the export group. -->
+    <export>...</export>
+    <exports>...</exports>
+  </exportGroup>
+  ```
```

Comment:
A real example please

Reply:
Copied from the documentation. 😄

Reply:
Oops! Then could we update the documentation as well? :sweat_smile:

Reply:
Oops, I should've left the comment after I pushed the change. Sorry. 😅